### PR TITLE
skip list added in c++

### LINKED ISSUE #1413: Skip List in C++

### PR DESCRIPTION
**Pull request by dldisha**
Fixes #1413

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Skip list added 

#### Languages Used:
- C++

#### Files Added:
- Skip_list/skip_list.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by dldisha** [comment]
@jainaman224, please review

File: Skip_list/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+**Time Complexity:-**
+
+| OPERATIONS | AVERAGE CASE | WORST CASE |
+| ---------- | ------------ | ---------- |
+| Search     | O(logn)	    | O(n)       |
+| Insertion  | O(logn)	    | O(n)       |
+| Deletion	 | O(logn)	    | O(n)       |
+
+
+
```
Comment (marwahmanbir) [inline Skip_list/README.md]: Remove the two extra blank lines above.
Rest, the README looks good :sparkles:

File: Skip_list/skip_list.cpp [hunk 2]
```diff
@@ -0,0 +1,340 @@
+#include <iostream>
+#include <cstdlib>
+#include <cmath>
+#include <cstring>
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove the space after ``include`` in the header files.

File: Skip_list/skip_list.cpp [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+#define MAX_LEVEL 6
+const float P = 0.5;
+using namespace std;
+
+//Skip Node Declaration
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Add a space after "//" in all comments.

File: Skip_list/skip_list.cpp [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+struct snode
+{
+    int value;
+    snode **forw;
+    snode (int level, int &value)
+    {
+        forw = new snode * [level + 1];
+        memset (forw, 0, sizeof(snode*) * (level + 1));
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove space after ``memset``.

File: Skip_list/skip_list.cpp [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+        this -> value = value; 
+    }
+    ~snode()
+    {
+        delete [] forw;        
+    } 
+};
+
+
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove one extra blank line.

File: Skip_list/skip_list.cpp [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+        cout << "2.Delete Element" << endl;
+        cout << "3.Search Element" << endl;
+        cout << "4.Display List " << endl;
+        cout << "5.Exit " << endl;
+        cout << "Enter your choice : ";
+        cin >> choice;
+        
+        switch (choice)
+        {
+
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove this blank line.

File: Skip_list/skip_list.cpp [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+             break;
+        
+        default:
+             cout << "Wrong Choice" << endl;
+        }
+    }
+    return 0;
+}
+ 
+
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove extra line.

File: Skip_list/skip_list.cpp [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+        }
+    }
+}
+ 
+//Delete Element from Skip List
+void skiplist::delete_element(int &value) 
+{
+    snode *x = header;	
+    snode *update[MAX_LEVEL + 1];
+    memset (update, 0, sizeof(snode*) * (MAX_LEVEL + 1));
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove space after memset.

File: Skip_list/skip_list.cpp [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+    for (int i = level; i >= 0; i--) 
+    {
+        while (x -> forw[i] != NULL && x -> forw[i] -> value < value)
```
Comment (marwahmanbir) [inline Skip_list/skip_list.cpp]: Remove space after ``while`` (in rest of the while statements also).

**Comment by dldisha** [comment]
@jainaman224 @rishabhgarg25699 @marwahmanbir can we now close and merge this PR

**Pull request merged**